asserts/checkers.go

```diff
@@ -92,3 +93,27 @@ func checkUint(headers map[string]string, name string, bitSize int) (uint64, err
 	}
 	return value, nil
 }
+
+func checkCommaSepList(headers map[string]string, name string) ([]string, error) {
+	listStr, ok := headers[name]
+	if !ok {
+		return nil, fmt.Errorf("%q header is mandatory", name)
+	}
```

Comment:
@emgee well this is a checker itself, so reuse is not so interesting, also I don't think we want many fields that must be there but can be empty, lists are a bit of a special case, maybe we even want to leave them out when empty but not sure about that